crates/red_knot_vendored/build.rs

```diff
@@ -55,21 +57,32 @@ fn zip_dir(directory_path: &str, writer: File) -> ZipResult<File> {
         // Some unzip tools unzip files with directory paths correctly, some do not!
         if absolute_path.is_file() {
             println!("adding file {absolute_path:?} as {normalized_relative_path:?} ...");
-            zip.start_file(normalized_relative_path, options)?;
+            zip.start_file(&*normalized_relative_path, options)?;
             let mut f = File::open(absolute_path)?;
             std::io::copy(&mut f, &mut zip).unwrap();
+
+            // Patch the VERSIONS file to make `knot_extensions` available
+            if normalized_relative_path == "stdlib/VERSIONS" {
+                writeln!(&mut zip, "knot_extensions: 3.0-")?;
+            }
```

Comment:
When I originally wrote this function, I intended to write it so that it could be used to zip up any directory into a zip archive -- that's why it's called `zip_dir()`, and that's why it accepts the typeshed directory path as a parameter rather than hardcoding it in the function. I think since I originally wrote it, it's already grown a few features that make it quite specific to its use-case here, such as the conditional logic a few lines above for deciding what the compression method should be. So I don't really mind much hardcoding things like `"stdlib/VERSIONS"` directly into the function body. But at this stage, it probably doesn't make much sense for the typeshed directory to be passed into the function as a parameter, and we should probably change the name of the function 😄

Reply:
Sorry about that, I figured it would be okay to modify in this way since it wasn't being used anywhere else and not in a place where it could be imported from anywhere else. See https://github.com/astral-sh/ruff/pull/15372